server/src/test/java/org/elasticsearch/gateway/PersistedClusterStateServiceTests.java

```diff
@@ -653,6 +654,27 @@ public void testPersistsAndReloadsIndexMetadataIffVersionOrTermChanges() throws
         }
     }
 
+    public void test_persists_updated_column_oid() throws IOException {
+        try (NodeEnvironment nodeEnvironment = newNodeEnvironment(createDataPaths())) {
+            final PersistedClusterStateService persistedClusterStateService = newPersistedClusterStateService(nodeEnvironment);
+            final long columnOid = 123L;
+
+            try (Writer writer = persistedClusterStateService.createWriter()) {
+                // Need to write full state first before doing incremental writes.
+                writer.writeFullStateAndCommit(1L, ClusterState.EMPTY_STATE);
+
+                // Only scenario when the metadata is written incrementally, had a bug.
```

Comment:
I'd remove this comment, it's a bit confusing. The test name already express what is tested.